native/src/components/TtsContainer.tsx

```diff
@@ -0,0 +1,181 @@
+  sentences: string[] | null
+  setSentences: (sentences: string[]) => void
+}
+
+export const ttsContext = createContext<TtsContextType>({
+  enabled: false,
+  setEnabled: () => undefined,
+  visible: false,
+  setVisible: () => undefined,
+  sentences: [],
+  setSentences: () => undefined,
+})
+
+type TtsContainerProps = {
+  children: ReactElement
+}
+
+const TtsContainer = ({ children }: TtsContainerProps): ReactElement => {
+  const { languageCode } = React.useContext(AppContext)
+  const { t } = useTranslation('layout')
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [sentenceIndex, setSentenceIndex] = useState(0)
+  const [visible, setVisible] = useState(false)
+  const [enabled, setEnabled] = useState(false)
+  const [sentences, setSentences] = useState<string[]>([])
+  const title = sentences[0] || t('nothingToRead')
+  const longTitle = truncate(title, { maxChars: MAX_TITLE_DISPLAY_CHARS })
+
+  const initializeTts = useCallback((): void => {
+    Tts.getInitStatus().catch(async error => {
+      reportError(`Tts-Error: ${error.code}`)
+      if (error.code === 'no_engine') {
+        await Tts.requestInstallEngine().catch((e: string) => reportError(`Failed to install tts engine: : ${e}`))
+      }
+    })
+  }, [])
+
+  const play = useCallback(
+    (index = sentenceIndex) => {
+      Tts.stop()
+      const sentence = sentences[index]
+      if (sentence) {
+        setIsPlaying(true)
+        Tts.speak(sentence, {
+          androidParams: {
+            KEY_PARAM_PAN: 0,
+            KEY_PARAM_VOLUME: 0.6,
+            KEY_PARAM_STREAM: 'STREAM_MUSIC',
+          },
+          iosVoiceId: '',
+          rate: 1,
+        })
+      }
+    },
+    [sentenceIndex, sentences],
+  )
+
+  const stop = async () => {
+    setIsPlaying(false)
+    setSentenceIndex(0)
+    await Tts.stop()
+    const TTS_STOP_DELAY = 100
+    await new Promise(resolve => {
+      setTimeout(resolve, TTS_STOP_DELAY)
+    })
+  }
+
+  const pause = () => {
+    Tts.stop()
+    setIsPlaying(false)
+  }
+
+  const playNext = useCallback(
+    (index = sentenceIndex) => {
+      const nextIndex = index + 1
+      if (nextIndex < sentences.length) {
+        setSentenceIndex(nextIndex)
+        play(nextIndex)
+      } else {
+        stop()
+      }
+    },
+    [play, sentenceIndex, sentences.length],
+  )
+
+  const playPrevious = (index = sentenceIndex) => {
+    const prevIndex = Math.max(0, index - 1)
+    if (prevIndex < sentences.length) {
+      setSentenceIndex(prevIndex)
+      play(prevIndex)
+    }
+  }
+
+  useEffect(() => {
+    if (!enabled) {
+      return () => undefined
+    }
+
+    initializeTts()
+    Tts.addEventListener('tts-finish', () => playNext(sentenceIndex))
+
+    return () => {
+      Tts.removeAllListeners('tts-finish')
+    }
+  }, [enabled, initializeTts, playNext, sentenceIndex, sentences.length])
+
+  useEffect(() => {
+    if (sentences.length <= 1) {
+      stop()
+    }
+  }, [sentences])
```

Comment:
:wrench: This `useEffect` should be removed by only passing a modified `setSentences` down that both sets the sentences and stops playing.